src/implementations/drip-querier.ts

```diff
@@ -22,15 +20,13 @@ import {
   VaultProtoConfigDoesNotExistError,
 } from '../errors';
 import Decimal from 'decimal.js';
+import { Config } from '../config';
 
 export class DripQuerierImpl implements DripQuerier {
   private readonly vaultProgram: Program<Drip>;
-  private readonly programId: PublicKey;
 
-  constructor(provider: AnchorProvider, private readonly network: Network, programId?: PublicKey) {
-    const config = Configs[network];
-    this.programId = programId ?? config.defaultProgramId;
-    this.vaultProgram = new Program(DripIDL as unknown as Drip, this.programId, provider);
+  constructor(provider: AnchorProvider, private readonly config: Config) {
+    this.vaultProgram = new Program(DripIDL as unknown as Drip, config.programId, provider);
```

Comment:
You can probably just import the IDL from the ./idl/type file and avoid this casting